### PR TITLE
creacion de comparacion del nit con la bd

### PR DESCRIPTION
**Pull request by david0110**

### CODE REVIEW

File: Back/Reciplastk.Gateway/Controllers/CustomerController.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public class CustomerController : ControllerBase
+    {
+        private readonly CustumerService custumerService;
+
+        public CustomerController(CustumerService custumerService)
+        {
+            this.custumerService = custumerService;
+        }
+
+        [HttpGet("Show all customers")]
```
Comment (PedroOspina17) [inline Back/Reciplastk.Gateway/Controllers/CustomerController.cs:19]: Remember, what you put inside the "" is the Value that goes in the URL. Do not include spaces or strange values. 
```suggestion
        [HttpGet("ShowAll")]
```

File: Back/Reciplastk.Gateway/Controllers/CustomerController.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        public HttpResponseModel ShowAllCustomers() {
+            var response = custumerService.ShowAllCustomers();
+            return response;
+        }
+
+        [HttpGet("Show a customer")]
```
Comment (PedroOspina17) [inline Back/Reciplastk.Gateway/Controllers/CustomerController.cs:25]:
```suggestion
        [HttpGet("Find")]
```

File: Back/Reciplastk.Gateway/Controllers/CustomerController.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        public HttpResponseModel ShowCustomer(int id)
+        {
+            var response = custumerService.ShowCustomer(id);
+            return response;
+        }
+        
+        [HttpPost("Create customer")]
```
Comment (PedroOspina17) [inline Back/Reciplastk.Gateway/Controllers/CustomerController.cs:32]:
```suggestion
        [HttpPost("Create")]
```

File: Back/Reciplastk.Gateway/Controllers/CustomerController.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        public HttpResponseModel CreateCustomer(CustomerViewModel customerViewModel)
+        {
+            var response = custumerService.CreateCustomer(customerViewModel);
+            return response;
+        }
+
+        [HttpPost("Edit customer")]
```
Comment (PedroOspina17) [inline Back/Reciplastk.Gateway/Controllers/CustomerController.cs:39]:
```suggestion
        [HttpPost("Edit")]
```

File: Back/Reciplastk.Gateway/Controllers/CustomerController.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        public HttpResponseModel EditCustomer(CustomerViewModel customerViewModel)
+        {
+            var response = custumerService.EditCustomer(customerViewModel);
+            return response;
+        }
+
+        [HttpDelete("Delete customer")]
```
Comment (PedroOspina17) [inline Back/Reciplastk.Gateway/Controllers/CustomerController.cs:46]:
```suggestion
        [HttpDelete("Delete")]
```

File: Back/Reciplastk.Gateway/Services/CustumerService.cs [hunk 6]
```diff
@@ -0,0 +1,115 @@
+﻿using Azure;
+using Reciplastk.Gateway.DataAccess;
+using Reciplastk.Gateway.Models;
+using System.Linq;
+
+namespace Reciplastk.Gateway.Services
+{
+    public class CustumerService
```
Comment (PedroOspina17) [inline Back/Reciplastk.Gateway/Services/CustumerService.cs:8]: The name is wrong. 
```suggestion
    public class CustomerService
```

File: Back/Reciplastk.Gateway/Services/CustumerService.cs [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+                response.Data = customer;
+            }
+            else
+            {
+                response.WasSuccessful = false;
+                response.StatusMessage = "The customer was not found";
+            }
+            return response;
+        }
+        private Customer CustomerById(int? customerId)
```
Comment (PedroOspina17) [inline Back/Reciplastk.Gateway/Services/CustumerService.cs:41]: Remember to name the methods starting with a verb. 
```suggestion
        private Customer GetCustomerById(int? customerId)
```

File: Back/Reciplastk.Gateway/Services/CustumerService.cs [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        {
+            var customer = db.Customers.FirstOrDefault(x => x.Customerid == customerId);
+            return customer;
+        }
+        private Customer CustomerByNit(String customerNit)
```
Comment (PedroOspina17) [inline Back/Reciplastk.Gateway/Services/CustumerService.cs:46]:
```suggestion
        private Customer GetCustomerByNit(String customerNit)
```

File: Back/Reciplastk.Gateway/Services/CustumerService.cs [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+                newCustomer.Clientsince = customerViewModel.Clientsince;
+                newCustomer.Needspickup = customerViewModel.Needspickup;
+                db.Customers.Add( newCustomer );
+                db.SaveChanges();
+                response.WasSuccessful = true;
+                response.Data = newCustomer;
+            }else
+            {
+                response.WasSuccessful = false;
+                response.StatusMessage = "The customer was not created";
```
Comment (PedroOspina17) [inline Back/Reciplastk.Gateway/Services/CustumerService.cs:71]: Remember to be more specific with the response messages. Also remember, the UI and messages that are meant to be shown in the UI are in Spanish for now. 
```suggestion
                response.StatusMessage = "Ya existe otro cliente con el NIT indicado";
```

File: Back/Reciplastk.Gateway/Services/CustumerService.cs [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+                customer.Clientsince = customerViewModel.Clientsince;
+                customer.Needspickup = customerViewModel.Needspickup;
+                db.SaveChanges();
+                response.WasSuccessful = true;
+                response.Data = customer;
+            }
+            else
+            {
+                response.WasSuccessful = false;
+                response.StatusMessage = "The customer was not found";
```
Comment (PedroOspina17) [inline Back/Reciplastk.Gateway/Services/CustumerService.cs:94]:
```suggestion
                response.StatusMessage = "No existe ningun cliente con el Id especificado";
```

File: Back/Reciplastk.Gateway/Services/CustumerService.cs [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+            return response;
+        }
+        public HttpResponseModel DeleteCustomer(int customerId) {
+            var response = new HttpResponseModel();
+            var customer = CustomerById(customerId);
+            if (customer != null) {
+                db.Remove(customer);
+                db.SaveChanges();
+                response.WasSuccessful = true;
+                response.StatusMessage = "The customer was deleted successfully";
```
Comment (PedroOspina17) [inline Back/Reciplastk.Gateway/Services/CustumerService.cs:105]:
```suggestion
                response.StatusMessage = "El cliente fue eliminado exitosamente";
```

File: Back/Reciplastk.Gateway/Services/CustumerService.cs [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+            }
+            else
+            {
+                response.WasSuccessful = false;
+                response.StatusMessage = "The customer was not deleted";
```
Comment (PedroOspina17) [inline Back/Reciplastk.Gateway/Services/CustumerService.cs:110]:
```suggestion
                response.StatusMessage = "No se encontró ningún cliente con el Id especificado";
```

**Review by PedroOspina17 — CHANGES_REQUESTED** [review]
Just a couple of comments, please adjust acordingly.

**Comment by david0110** [comment]
ya fueron realizadas todas las correciones

**Pull request closed without merging**